In [12]:
import pandas as pd
import matplotlib.pyplot as plt

communes_2022 = pd.read_csv("../data/processed/communes_2022.csv")
display(communes_2022.shape)
display(communes_2022.head())
communes_2022.isna().sum()

(2115, 23)

,code,name,population,foreigners,foreign_share,share_under20,share_65plus,density,canton,major_region,...,sa_rate,jobs_primary,jobs_secondary,jobs_tertiary,jobs_total,tax_normal,tax_special,tax_total,tax_per_capita,jobs_per_capita
0,1,Aeugst am Albis,1991,294.0,14.8,18.8,20.4,251.7,ZH,Zurich,...,0.9,42.0,18.0,410.0,470.0,4821708.65,902644.05,5724352.70,2875.114365,0.236062
1,10,Obfelden,5787,1242.0,21.5,23.5,17.5,767.5,ZH,Zurich,...,2.4,53.0,284.0,1129.0,1466.0,5790084.95,591236.30,6381321.25,1102.699369,0.253326
2,100,Stadel,2349,317.0,13.5,20.1,17.8,182.2,ZH,Zurich,...,1.3,92.0,63.0,404.0,559.0,2289562.55,172083.85,2461646.40,1047.955045,0.237974
3,1001,Doppleschwand,806,34.0,4.2,24.3,15.8,116.0,LU,Central CH,...,NaN,76.0,49.0,94.0,219.0,362504.40,5514.00,368018.40,456.598511,0.271712
4,1002,Entlebuch,3267,231.0,7.1,21.2,19.0,57.4,LU,Central CH,...,0.8,385.0,481.0,1045.0,1911.0,1466934.15,293908.15,1760842.30,538.978359,0.584940


code                 0
name                 0
population           0
foreigners           6
foreign_share        6
share_under20        2
share_65plus         0
density              0
canton               1
major_region         1
urban_rural          1
language             1
sa_recipients      487
sa_rate            487
jobs_primary        50
jobs_secondary      67
jobs_tertiary       14
jobs_total           3
tax_normal          10
tax_special         10
tax_total           10
tax_per_capita      10
jobs_per_capita      3
dtype: int64

# Terminology clarification

- **Commune**: one row of the table; Swiss communes on the 2025 list (2,115). Data refer to 2022.
- **Population**: permanent resident population of the commune.
- **Foreigners**: permanent residents without Swiss citizenship.
- **Foreign share**: foreigners ÷ population, in %.
- **Share under 20 / share 65+**: % of residents in each age group.
- **Density**: residents per km² of total commune area.

**Outcomes**
- **Tax per resident**: federal income tax only (not cantonal or communal), per year, divided by all residents, including children and people who pay nothing.
  - *Normal cases*: taxpayers taxed normally for the full year.
  - *Special cases*: e.g. people who moved in or out during the year, or are only partly taxable in Switzerland.
- **Social assistance rate**: % of residents who receive social assistance (last-resort cash support), not an amount of money.
- **Social assistance recipients**: number of people receiving it.

**Economy**
- **Jobs**: jobs located in the commune (where people work, not where they live).
  - *Primary*: agriculture, forestry. *Secondary*: industry, construction. *Tertiary*: services.
- **Jobs per resident**: jobs ÷ population; above 1 means a job centre that attracts commuters.

**Classifications (SFSO, 2020)**
- **Canton**: one of 26.
- **Major region**: 7 large regions (e.g. Lake Geneva, Zurich).
- **Urban / intermediate / rural**: SFSO urban–rural typology.
- **Language region**: German, French, Italian or Romansh.

**Missing values**
- **NaN**: value not available, mostly hidden for privacy in small communes.